Update dispersion corrections tutorial for CASTEP 24.1

jkshenton opened the pull request:
- include D4 correction
- change guidance on supercell warnings for TS, MBD and XDM corrections
- Make the ASE-based script work with ASE 3.23

jkshenton committed:
Update dispersion corrections tutorial for CASTEP 24.1
- include D4 correction
- change guidance on supercell warnings for TS, MBD and XDM corrections
- Make the ASE-based script work with ASE 3.23

krefson commented on docs/tutorials/dispersion_corrections/castep-ase-dispersion-tutorial.md:
I suggest somewhere including a comment that most published results for "D3" are actually "D3-BJ", and that the original version of D3 is rarely used any more.

jkshenton replied:
I've added that note now to the main dispersion corrections page (where other similar recommendations were given).

jkshenton committed:
Added D4 to main dispersion docs
- also added note to say D3 without BJ correction is rarely used.
- Minor formatting tweaks to dispersion tutorial
- Added references for the D4 method.
